### ISSUE TITLE
YaCy crashes randomly under Debian9?

### ISSUE THREAD

**Issue by r4dh4l** [yacy/yacy_search_server#252]
Hi, 

I try to run a YaCy node in a headless Debian9 VM:

```
# uname -a
Linux vm-yacy 4.9.0-8-amd64 #1 SMP Debian 4.9.110-3+deb9u6 (2018-10-08) x86_64 GNU/Linux

# apt-cache policy *jre-headless | less
openjdk-8-jre-headless:
  Installed: 8u181-b13-2~deb9u1
  Candidate: 8u181-b13-2~deb9u1
  Version table:
...
 *** 8u181-b13-2~deb9u1 500
        500 http://ftp.de.debian.org/debian stretch/main amd64 Packages
        500 http://security.debian.org/debian-security stretch/updates/main amd64 Packages
        100 /var/lib/dpkg/status
...

# apt policy yacy
yacy:
  Installed: 1.921.9788
  Candidate: 1.921.9788
  Version table:
 *** 1.921.9788 500
        500 https://dl.bintray.com/luccioman/yacy_search_server stretch/main amd64 Packages
        100 /var/lib/dpkg/status
```

So currently I use https://github.com/yacy/yacy_search_server/issues/124#issuecomment-339051269 @luccioman because the official Debian package has the problem reported in https://github.com/yacy/yacy_search_server/issues/124 and https://yacy.net/release/yacy_v1.92_20161226_9000.tar.gz, which I tried first, crashed randomly as well (so this seems nothing specific for the package provided by @luccioman ).

Unfortunately `/var/log/yacyyacy00.log` does not report something suspicious - the log just ends when the java processes crash/end.

Any ideas why this happen or how I could better investigate this?

**Comment by luccioman** [comment]
Hi @r4dh4l , I regularly use and test YaCy on Debian 9 but until now I did not encounter random crashes, except when testing on JDK 11 (see issue #241).
Did you also try with the full jre or jdk package (not headless)?

**Comment by r4dh4l** [comment]
Thx for your reply @luccioman: No I didn't  try the non-headless versions yet sind http://wiki.yacy.net/index.php/En:DebianInstall says:

> ... only a headless version is needed

Well, I just noticed that the process runs stable as long nobody is searching something. I gave someone my server URL short after the user tried a search the process crashed. Any idea why this happens?

Anyway: Which version should I try:

```
# apt policy openjdk-8-jre*
openjdk-8-jre:
  Installed: (none)
  Candidate: 8u181-b13-2~deb9u1
  Version table:
     8u181-b13-2~deb9u1 500
        500 http://ftp.de.debian.org/debian stretch/main amd64 Packages
        500 http://security.debian.org/debian-security stretch/updates/main amd64 Packages
openjdk-8-jre-dcevm:
  Installed: (none)
  Candidate: 8u112-1
  Version table:
     8u181-1~bpo9+1 100
        100 http://ftp.debian.org/debian stretch-backports/main amd64 Packages
     8u112-1 500
        500 http://ftp.de.debian.org/debian stretch/main amd64 Packages
openjdk-8-jre-zero:
  Installed: (none)
  Candidate: 8u181-b13-2~deb9u1
  Version table:
     8u181-b13-2~deb9u1 500
        500 http://ftp.de.debian.org/debian stretch/main amd64 Packages
        500 http://security.debian.org/debian-security stretch/updates/main amd64 Packages
openjdk-8-jre-headless:
  Installed: 8u181-b13-2~deb9u1
  Candidate: 8u181-b13-2~deb9u1
  Version table:
 *** 8u181-b13-2~deb9u1 500
        500 http://ftp.de.debian.org/debian stretch/main amd64 Packages
        500 http://security.debian.org/debian-security stretch/updates/main amd64 Packages
        100 /var/lib/dpkg/status
# 
```
?

**Comment by luccioman** [comment]
Yes @r4dh4l YaCy is supposed to run with a headless jre, but your issue looks like the one that occurred with the JDK 11 (randomness, no trace of error in logs). That's why I wondered if the JVM version itself could be involved.
So I would check if the same problem occurs with openjdk-8-jre or openjdk-8-jdk packages.

**Comment by r4dh4l** [comment]
I think I found the problem: Because my Hypervisor has not so much RAM I've set a "ballooning" for the VM providing 512MB min. and 2GB max. Somehow this didn't work properly. Providing fixed 2GB RAM the yacy/java processes work fine until now.

**Comment by luccioman** [comment]
Ok @r4dh4l , yes when the memory requested by the YaCy process is coming close or beyond the limit dedicated to its containing virtual machine this is a possible cause of crash.
As far as I know it is not specific to YaCy but can occur with any Java program, so I think you can close the issue once you have been running YaCy without crash for a sufficiently long time.
